### PR TITLE
feat: add module deployment lib

### PR DESCRIPTION
**Pull request by kopy-kat**

### CODE REVIEW

File: src/modulekit/deployment/RegistryDeployer.sol [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    ModuleRecord,
+    IResolver,
+    ResolverBase,
+    DebugResolver,
+    ResolverRecord
+} from "../../test/utils/dependencies/Registry.sol";
+
+contract RegistryDeployer {
+    IRegistry registry = IRegistry(REGISTRY_ADDR);
+    bytes32 resolverUID = 0x984f176bc8a8b71d1a35736c5a892be396a01ba80b290a3394d0089b891dcf46;
```
Comment (zeroknots) [inline src/modulekit/deployment/RegistryDeployer.sol:17]: what is this resolverUID?

File: src/modulekit/deployment/RegistryDeployer.sol [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        returns (address moduleAddr)
+    {
+        bytes32 _resolverUID = getResolver();
+        moduleAddr = registry.deployViaFactory(factory, callOnFactory, data, _resolverUID);
+    }
+
+    // <---- REGISTRY MANAGEMENT ---->
+
+    function getResolver() public returns (bytes32 _resolverUID) {
+        _resolverUID = resolverUID;
```
Comment (zeroknots) [inline src/modulekit/deployment/RegistryDeployer.sol:63]: shouldn't the resolver UID be user provided? we are only supporting the hard-coded value here

File: src/modulekit/deployment/RegistryDeployer.sol [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    function deployModule(
+        bytes memory code,
+        bytes memory deployParams,
+        bytes32 salt,
+        bytes memory data
+    )
+        public
+        returns (address moduleAddr)
+    {
+        bytes32 _resolverUID = getResolver();
```
Comment (zeroknots) [inline src/modulekit/deployment/RegistryDeployer.sol:31]: dev cant chose resolver?!

**Review by zeroknots — COMMENTED** [review]
I think it would be better to build this as a library with user supplied params, so it can easily be included to 3rd party tool chains

File: src/modulekit/deployment/RegistryDeployer.sol (hunk 3 above)
Comment (kopy-kat) [reply src/modulekit/deployment/RegistryDeployer.sol:29]: setter added

File: src/modulekit/deployment/RegistryDeployer.sol (hunk 2 above)
Comment (kopy-kat) [reply src/modulekit/deployment/RegistryDeployer.sol:61]: see above

File: src/modulekit/deployment/RegistryDeployer.sol (hunk 1 above)
Comment (kopy-kat) [reply src/modulekit/deployment/RegistryDeployer.sol]: This is a default resolver for dev convenience

**Comment by zeroknots** [comment]
fixed conflict

**Pull request merged**